src/hotspot/cpu/x86/x86.ad

```diff
@@ -1868,10 +1868,11 @@ const bool Matcher::match_rule_supported_vector(int opcode, int vlen, BasicType
       }
       break;
     case Op_VectorCastD2X:
-      if (is_subword_type(bt) || bt == T_INT) {
+      if (((UseAVX <= 2) || (!VM_Version::supports_avx512vl())) &&
+          (is_subword_type(bt) || bt == T_INT)) {
         return false;
       }
-      if (bt == T_LONG && !VM_Version::supports_avx512dq()) {
+      if (is_integral_type(bt) && !VM_Version::supports_avx512dq()) {
         return false;
       }
```

Comment:
Overlapping conditions for the same types are confusing.

Reply:
I will add comments and rephrase the checks to make it clearer.